### PR TITLE
Add minor theme support

### PR DESCRIPTION
**Pull request by fdncred**
working except for table lines. I really have no idea why table lines aren't being colored. I've validated that the color style is set and I can write an output string with that color style in the assigned color however at runtime it doesn't color them. maybe a bug in my code or in ansi_term, who knows. would appreciate some eyes on this. i'm sure there will be lots of suggested changes.

changes to config.toml
```
[color_config]
primitive_int = "green"
primitive_decimal = "red"
primitive_filesize = "blue"
primitive_string = "purple"
primitive_line = "yellow"
primitive_columnpath = "cyan"
primitive_pattern = "white"
primitive_boolean = "green"
primitive_date = "red"
primitive_duration = "blue"
primitive_range = "purple"
primitive_path = "yellow"
primitive_binary = "cyan"
separator_color = "purple"
header_align = "l" # left|l, right|r, center|c
header_color = "c" # green|g, red|r, blue|u, black|b, yellow|y, purple|p, cyan|c, white|w
header_bold = true
header_style = "cb"
index_color = "r"
```
breaking changes to config.toml
moved `header_color`, `header_bold`, and `header_align` into the `color_config` group

still lots of cleanup to do. green code everywhere (comments). not very rust idomatic but considering a c# person wrote it, that is to be expected. :)`

### CODE REVIEW

**Comment by fdncred** [comment]
@jonathandturner This will probably need some work but if you want to get it in this release feel free to dive in, I can look at any comments pretty quick. Also, I think this only works on `ls` right now. I'll probably have to wait for an upgraded config system to color primitives in other places.

File: crates/nu-table/src/table.rs [hunk 1]
```diff
@@ -300,102 +504,195 @@ pub struct WrappedTable {
...
+                        .to_string(),
+                );
+            }
+
             for column in cells.iter().enumerate() {
                 if let Some(line) = (column.1).lines.get(current_line) {
                     let remainder = self.column_widths[column.0] - line.width;
                     output.push(' ');
 
+                    // println!("Column1: [{:?}]", column.1.style);
```
Comment (jonathandturner) [inline crates/nu-table/src/table.rs:691]: ditto (okay you get the picture 😄 )

**Review by jonathandturner — COMMENTED** [review]
Should be fine. We'll no doubt keep evolving this as we build out our theme support, but this seems like it's a good start in that direction.

**Comment by fdncred** [comment]
Ya, I hope to keep evolving it. I already have some ideas about enhancing this, for instance allow the configuration of `styles` instead of just colors. Styles have 10 properties so I'm not sure how to do that exactly. Maybe `fg|bg|1|1|0|0|1|1|0|0` or perhaps json.  Just not sure.

**Pull request merged**